apis/Google.Cloud.Spanner.Data/Google.Cloud.Spanner.Data/SpannerDbType.cs

```diff
@@ -12,66 +12,170 @@
+                default: return new V1.Type {Code = TypeCode};
+            }
+        }
+
         /// <summary>
         /// Not specified.
         /// </summary>
-        Unspecified = 0,
+        public static SpannerDbType Unspecified { get; } = new SpannerDbType(TypeCode.Unspecified);
 
         /// <summary>
         /// Encoded as JSON `true` or `false`.
         /// </summary>
-        Bool = 1,
+        public static SpannerDbType Bool { get; } = new SpannerDbType(TypeCode.Bool);
 
         /// <summary>
         /// Encoded as `string`, in decimal format.
         /// </summary>
-        Int64 = 2,
+        public static SpannerDbType Int64 { get; } = new SpannerDbType(TypeCode.Int64);
 
         /// <summary>
         /// Encoded as `number`, or the strings `"NaN"`, `"Infinity"`, or
         /// `"-Infinity"`.
         /// </summary>
-        Float64 = 3,
+        public static SpannerDbType Float64 { get; } = new SpannerDbType(TypeCode.Float64);
 
         /// <summary>
         /// Encoded as `string` in RFC 3339 timestamp format. The time zone
         /// must be present, and must be `"Z"`.
         /// </summary>
-        Timestamp = 4,
+        public static SpannerDbType Timestamp { get; } = new SpannerDbType(TypeCode.Timestamp);
 
         /// <summary>
         /// Encoded as `string` in RFC 3339 date format.
         /// </summary>
-        Date = 5,
+        public static SpannerDbType Date { get; } = new SpannerDbType(TypeCode.Date);
 
         /// <summary>
         /// Encoded as `string`.
         /// </summary>
-        String = 6,
+        public static SpannerDbType String { get; } = new SpannerDbType(TypeCode.String);
 
         /// <summary>
         /// Encoded as a base64-encoded `string`, as described in RFC 4648,
         /// section 4.
         /// </summary>
-        Bytes = 7,
+        public static SpannerDbType Bytes { get; } = new SpannerDbType(TypeCode.Bytes);
 
         /// <summary>
         /// Encoded as `list`, where the list elements are represented
         /// according to [array_element_type][google.spanner.v1.Type.array_element_type].
         /// </summary>
-        Array = 8,
+        public static SpannerDbType ArrayOf(SpannerDbType elementType) => new SpannerDbType(TypeCode.Array, elementType);
 
         /// <summary>
         /// Encoded as `list`, where list element `i` is represented according
         /// to [struct_type.fields[i]][google.spanner.v1.StructType.fields].
         /// </summary>
-        Struct = 9
+        public static SpannerDbType StructOf(IDictionary<string, SpannerDbType> structMembers) => new SpannerDbType(TypeCode.Struct, structMembers);
+
+        internal TypeCode TypeCode { get; }
+
+        internal SpannerDbType ArrayElementType { get; }
+
+        internal IDictionary<string, SpannerDbType> StructMembers { get; }
+
+        /// <inheritdoc />
+        public override bool Equals(object obj)
+        {
+            return Equals(obj as SpannerDbType);
```

Comment:
let me do that in the resharper pass as I have seen a few other places where this could be optimized.
to be clear, we want to use expression bodied methods only if its a single line right?